src/mf6sub.f90

```diff
@@ -0,0 +1,239 @@
+                                    basemodellist, baseexchangelist,           &
+                                    lists_da
+  use SimVariablesModule,     only: iout 
+  use SimModule,              only: converge_reset, converge_check,            &
+                                    final_message
+  use TdisModule,             only: tdis_tu, tdis_da,                          &
+                                    endofsimulation
+  implicit none
+
+  ! -- dummy
+  external :: mf6cback
+  
+  ! -- local
+  class(SolutionGroupType), pointer :: sgp => null()
+  class(BaseSolutionType), pointer :: sp => null()
+  class(BaseModelType), pointer :: mp => null()
+  class(BaseExchangeType), pointer :: ep => null()
+  integer(I4B) :: im, ic, is, isg
+  logical :: exit_tsloop
+  character(len=80) :: compiler
+  ! -- formats
+! ------------------------------------------------------------------------------
+  !
+  ! -- parse any command line arguments
+  call GetCommandLineArguments()
+  !
+  ! -- Write banner to screen (unit 6) and start timer
+  call write_centered('MODFLOW'//MFVNAM, ISTDOUT, 80)
+  call write_centered(MFTITLE, ISTDOUT, 80)
+  call write_centered('VERSION '//VERSION, ISTDOUT, 80)
+  !
+  ! -- Write if develop mode
+  if (IDEVELOPMODE == 1) call write_centered('***DEVELOP MODE***', ISTDOUT, 80)
+  !
+  ! -- Write compiler version
+  call get_compiler(compiler)
+  call write_centered(' ', ISTDOUT, 80)
+  call write_centered(trim(adjustl(compiler)), ISTDOUT, 80)
+  !
+  ! -- Write disclaimer
+  write(ISTDOUT, FMTDISCLAIMER)
+  ! -- get start time
+  call start_time()
+  !
+  !
+  ! -- CREATE (CR)
+  call simulation_cr()
+  !
+  !
+  ! -- DEFINE (DF)
+  ! -- Define each model
+  do im = 1, basemodellist%Count()
+    mp => GetBaseModelFromList(basemodellist, im)
+    call mp%model_df()
+  enddo
+  !
+  ! -- Define each exchange
+  do ic = 1, baseexchangelist%Count()
+    ep => GetBaseExchangeFromList(baseexchangelist, ic)
+    call ep%exg_df()
+  enddo
+  !
+  ! -- Define each solution
+  do is = 1, basesolutionlist%Count()
+    sp => GetBaseSolutionFromList(basesolutionlist, is)
+    call sp%sln_df()
+  enddo
+  !
+  !
+  ! -- ALLOCATE AND READ (AR)
+  ! -- Allocate and read each model
+  do im = 1, basemodellist%Count()
+    mp => GetBaseModelFromList(basemodellist, im)
+    call mp%model_ar()
+  enddo
+  !
+  ! -- Allocate and read each exchange
+  do ic = 1, baseexchangelist%Count()
+    ep => GetBaseExchangeFromList(baseexchangelist, ic)
+    call ep%exg_ar()
+  enddo
+  !
+  ! -- Allocate and read each solution
+  do is=1,basesolutionlist%Count()
+    sp => GetBaseSolutionFromList(basesolutionlist, is)
+    call sp%sln_ar()
+  enddo
+  !
+  !
+  ! -- TIME STEP LOOP
+  tsloop: do
+    !
+    ! -- TIME UPDATE (TU)
+    call tdis_tu()
+    !
+    ! -- MODFLOW 6 callback function allows an external program
+    !    to modify program during execution
+    call mf6cback()
+    !

```

Comment:
Please move `call mf6cback()` to line 143, i.e. after:

        do is=1,basesolutionlist%Count()
          sp => GetBaseSolutionFromList(basesolutionlist, is)
          call sp%sln_rp()
        enddo